### ISSUE TITLE
coherence between the opcode ad the operand

### ISSUE THREAD

**Issue by ankilla** [sailro/Reflexil#10]
Hi

There is function 
<b>
reducehealth(int damage, gameobject killer)
{
health = health-damage
}
</b>
this function is used by both ai & player so i cant make (health=health+damage) which effects both player as well as ai. so I am adding as
<b>
if(gameobject='player')
{
damage = 0
}
</b>
used op code is as follows
<b>
Opcode : ldc.i4.s
Operand type: ->Parameter reference
Operand : ->(0)damage (system.int32)
</b>
when adding this instruction reflexil is throwing error as
<b>"Reflexil is unable to create this instruction, check coherence between the opcode ad the operand"</b>

how to do this.

**Comment by sailro** [comment]
ldc.i4.s is to be used with a const integer.

Setting argument damage = 0, in IL is:
opcode ldc.i4.0 (no argument)
opcode starg.s, optype Parameter Reference, operand (0)damage (system.int32) 

You should use "replace all with code" feature if you are not comfortable with IL.

**Comment by ankilla** [comment]
I tried  "replace all with code" feature but i am getting error as 
<b>The type of namespace name 'UnityEngine' could not be found (are you missing a using directive or an assembly reference?)</b>
I tried
<b>using UnityEngine;</b>
then also giving the same error

**Comment by sailro** [comment]
If you can send me the assembly and the method you are trying to change, it would be great.

**Comment by ankilla** [comment]
Hi

Actually I do game hacking and post on domain, the main issue is
sitelocking the hacked game, while checking on this I found a site where
they are doing this. When I checked the assembly I found they had injected
3 classes (Hgbehaviour, Hgutils, Hgtrack) using sitelock method in Hgutils
class they are sitelocking.

When I also injected a class and created a method when adding code
"Replace" I am getting the error.

I am attaching the sitelcoked assembly for your ref & the assembly which I
want to modify.

The game on which i am working is 'Broforce'

Thanks
Ankilla
